src/kv/kv.h

```diff
@@ -1035,9 +1033,11 @@ namespace kv
               // because of this we are returning NoVersion
               h->add_result(req_id, NoVersion);
             }
+            committed = true;
```

Comment:
I think we should just set this once, at line 1018. And we should also move the similar line in `commit_reserved`, to be after we've retrieved a commit version.